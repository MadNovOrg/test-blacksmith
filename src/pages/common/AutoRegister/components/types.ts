import { schemas, yup, TFunction } from '@app/schemas'
import { requiredMsg } from '@app/util'

export type FormInputs = {
  firstName: string
  surname: string
  marketing: boolean
  phone: string
  password: string
  dob: Date | null
  tcs: boolean
}

export const getFormSchema = (t: TFunction) => {
  return yup.object({
    firstName: yup.string().required(requiredMsg(t, 'first-name')),
    surname: yup.string().required(requiredMsg(t, 'surname')),
    password: schemas.password(t),

    phone: yup.string().required(requiredMsg(t, 'phone')),
    dob: yup
      .date()
      .typeError(t('validation-errors.invalid-date-optional'))
      .nullable(),

    marketing: yup.boolean(),
    tcs: yup.boolean().oneOf([true], t('pages.signup.tcs-required')),
  })
}
