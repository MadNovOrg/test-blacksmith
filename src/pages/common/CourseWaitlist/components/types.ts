import { schemas, yup, TFunction } from '@app/schemas'
import { requiredMsg } from '@app/util'

export const getFormSchema = (t: TFunction) => {
  return yup.object({
    firstName: yup.string().required(requiredMsg(t, 'first-name')),

    surname: yup.string().required(requiredMsg(t, 'surname')),

    email: schemas.email(t, true),

    phone: yup
      .string()
      .required(requiredMsg(t, 'phone'))
      .min(10, t('validation-errors.phone-invalid')),

    orgName: yup.string().required(requiredMsg(t, 'org-name')),
  })
}
