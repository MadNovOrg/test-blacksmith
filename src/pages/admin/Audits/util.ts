import {
  Course_Type_Enum,
  GetAttendeeAuditLogsQuery,
  GetCourseAuditLogsQuery,
} from '@app/generated/graphql'

export const getExportDataRenderFunction = <T>(
  cols: { label: string; exportRender: (o: T) => string }[],
  logs: T[]
) => {
  return () => {
    return [
      cols.map(c => c.label),
      ...logs.map(row => cols.map(col => col.exportRender(row))),
    ]
  }
}

export type CourseLogType = GetCourseAuditLogsQuery['logs'][0]
export type AttendeeLogType = GetAttendeeAuditLogsQuery['logs'][0]

export const getCourseInvoice = (log: CourseLogType) => {
  if (
    log.course.type === Course_Type_Enum.Closed ||
    log.course.type === Course_Type_Enum.Indirect
  ) {
    if (log.course.orders.length > 0) {
      return log.course.orders[0]
    }
  }
  return null
}
export const getAttendeeInvoice = (log: AttendeeLogType) => {
  if (
    log.course.type === Course_Type_Enum.Closed ||
    log.course.type === Course_Type_Enum.Indirect
  ) {
    if (log.course.orders.length > 0) {
      return log.course.orders[0]
    }
  } else {
    return log.course.orders.find(
      order => order.registrants.indexOf(log.profile.email) > -1
    )
  }
  return null
}
