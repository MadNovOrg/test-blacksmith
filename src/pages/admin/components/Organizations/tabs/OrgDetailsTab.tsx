import { Box, Button, CircularProgress, Grid, Stack } from '@mui/material'
import Typography from '@mui/material/Typography'
import { format } from 'date-fns-tz'
import React from 'react'
import { useNavigate } from 'react-router-dom'

import { DetailsRow } from '@app/components/DetailsRow'
import { useAuth } from '@app/context/auth'
import useOrg from '@app/hooks/useOrg'
import { useScopedTranslation } from '@app/hooks/useScopedTranslation'
import { sectors } from '@app/pages/common/CourseBooking/components/org-data'
import { LoadingStatus } from '@app/util'

type OrgDetailsTabParams = {
  orgId: string
}

export const OrgDetailsTab: React.FC<
  React.PropsWithChildren<OrgDetailsTabParams>
> = ({ orgId }) => {
  const { t, _t } = useScopedTranslation('pages.org-details.tabs.details')
  const { profile, acl } = useAuth()
  const navigate = useNavigate()

  const { data, status } = useOrg(
    orgId,
    profile?.id,
    acl.canViewAllOrganizations()
  )

  const org = data?.length ? data[0] : null
  const [, sector = org?.sector] =
    Object.entries(sectors).find(
      ([key, value]) => key === org?.sector && value
    ) || []
  return (
    <Box sx={{ pt: 2, pb: 4 }}>
      {status === LoadingStatus.FETCHING ? (
        <Stack
          alignItems="center"
          justifyContent="center"
          data-testid="org-details-fetching"
        >
          <CircularProgress />
        </Stack>
      ) : null}

      {org && status === LoadingStatus.SUCCESS ? (
        <Box>
          <Typography variant="subtitle1" mb={2}>
            {t('organization-details-section.title')}
          </Typography>

          <Box bgcolor="common.white" p={3} mb={4} borderRadius={1}>
            <Grid container>
              <Grid item xs={8}>
                <DetailsRow
                  label={t('organization-details-section.organization-name')}
                  value={org.name}
                />
                <DetailsRow
                  label={t('organization-details-section.organization-sector')}
                  value={sector || ''}
                />
                <DetailsRow
                  label={t('organization-details-section.organization-type')}
                  value={org.organisationType}
                />
                <DetailsRow
                  label={t('organization-details-section.organization-phone')}
                  value={org.attributes.phone}
                />
                <DetailsRow
                  label={t('organization-details-section.organization-email')}
                  value={org.attributes.email}
                />
                <DetailsRow
                  label={t('organization-details-section.organization-website')}
                  value={org.attributes.website}
                />
              </Grid>

              {acl.canEditOrgs() ? (
                <Grid item xs={4}>
                  <Box display="flex" justifyContent="flex-end">
                    <Button
                      variant="outlined"
                      color="primary"
                      size="large"
                      onClick={() => navigate('./edit')}
                    >
                      {_t('pages.org-details.edit-organization')}
                    </Button>
                  </Box>
                </Grid>
              ) : null}
            </Grid>
          </Box>

          <Typography variant="subtitle1" mb={2}>
            {t('organization-address-section.title')}
          </Typography>

          <Box bgcolor="common.white" p={3} mb={4} borderRadius={1}>
            <DetailsRow
              label={t('organization-address-section.address-line-1')}
              value={org.address.line1}
            />
            <DetailsRow
              label={t('organization-address-section.address-line-2')}
              value={org.address.line2}
            />
            <DetailsRow
              label={t('organization-address-section.town-or-city')}
              value={org.address.city}
            />
            <DetailsRow
              label={t('organization-address-section.postcode')}
              value={org.address.postCode}
            />
            <DetailsRow
              label={t('organization-address-section.country')}
              value={org.address.country}
            />
          </Box>

          <Typography variant="subtitle1" mb={2}>
            {t('additional-details.title')}
          </Typography>

          <Box bgcolor="common.white" p={3} mb={4} borderRadius={1}>
            <DetailsRow
              label={t('additional-details.head-first-name')}
              value={org.attributes.headFirstName}
            />
            <DetailsRow
              label={t('additional-details.head-surname')}
              value={org.attributes.headSurname}
            />
            <DetailsRow
              label={t('additional-details.head-email')}
              value={org.attributes.headEmailAddress}
            />
            <DetailsRow
              label={t('additional-details.setting')}
              value={org.attributes.settingName}
            />
            <DetailsRow
              label={t('additional-details.local-authority')}
              value={org.attributes.localAuthority}
            />
            <DetailsRow
              label={t('additional-details.ofsted-rating')}
              value={org.attributes.ofstedRating}
            />
            <DetailsRow
              label={t('additional-details.ofsted-last-inspection')}
              value={
                org.attributes.ofstedLastInspection
                  ? format(
                      new Date(org.attributes.ofstedLastInspection),
                      'd MMMM yyyy'
                    )
                  : null
              }
            />
          </Box>
        </Box>
      ) : null}
    </Box>
  )
}
