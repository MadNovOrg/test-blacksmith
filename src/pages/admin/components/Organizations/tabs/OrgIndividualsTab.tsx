import PersonAddIcon from '@mui/icons-material/PersonAdd'
import { TabContext, TabList, TabPanel } from '@mui/lab'
import {
  Box,
  Button,
  CircularProgress,
  Container,
  Grid,
  Stack,
  Tab,
} from '@mui/material'
import Typography from '@mui/material/Typography'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useNavigate, useSearchParams } from 'react-router-dom'

import { OrgInvitesTable } from '@app/components/OrgInvitesTable'
import { OrgUsersTable } from '@app/components/OrgUsersTable'
import { useAuth } from '@app/context/auth'
import useOrg from '@app/hooks/useOrg'
import { LoadingStatus } from '@app/util'

export enum OrgIndividualsSubtabs {
  USERS = 'USERS',
  INVITES = 'INVITES',
}

type OrgIndividualsTabParams = {
  orgId: string
}

export const OrgIndividualsTab: React.FC<OrgIndividualsTabParams> = ({
  orgId,
}) => {
  const { t } = useTranslation()
  const { profile } = useAuth()
  const navigate = useNavigate()
  const [searchParams] = useSearchParams()

  const initialTab = searchParams.get('subtab') as OrgIndividualsSubtabs | null
  const [selectedTab, setSelectedTab] = useState(
    initialTab || OrgIndividualsSubtabs.USERS
  )

  useEffect(() => {
    if (initialTab) setSelectedTab(initialTab)
  }, [initialTab])

  const { data, stats, status, mutate } = useOrg(orgId, profile?.id)

  const org = data?.length ? data[0] : null

  return (
    <Container maxWidth="lg" sx={{ pt: 2, pb: 4 }}>
      {status === LoadingStatus.FETCHING ? (
        <Stack
          alignItems="center"
          justifyContent="center"
          data-testid="org-users-fetching"
        >
          <CircularProgress />
        </Stack>
      ) : null}

      {org && status === LoadingStatus.SUCCESS ? (
        <>
          <Grid container>
            <Grid item xs={4} bgcolor="common.white" p={3} borderRadius={1}>
              <Typography variant="h2" mb={1}>
                {stats.profiles.count}
              </Typography>
              <Typography variant="body2" mb={1}>
                {t(
                  'pages.org-details.tabs.details.user-details.total-number-of-users'
                )}
              </Typography>
            </Grid>

            <Grid item xs={4} bgcolor="common.white" p={3} borderRadius={1}>
              <Typography variant="h2" mb={1}>
                {stats.certificates.active.count}
              </Typography>
              <Typography variant="body2" mb={1}>
                {t(
                  'pages.org-details.tabs.details.user-details.active-certifications'
                )}
              </Typography>
            </Grid>

            <Grid item xs={4} bgcolor="common.white" p={3} borderRadius={1}>
              <Typography variant="h2" mb={1}>
                {stats.certificates.expired.count}
              </Typography>
              <Typography variant="body2" mb={1}>
                {t(
                  'pages.org-details.tabs.details.user-details.expired-certifications'
                )}
              </Typography>
            </Grid>
          </Grid>

          <Box mt={2}>
            <TabContext value={selectedTab}>
              <Box display="flex" justifyContent="space-between" my={2}>
                <TabList onChange={(_, value) => setSelectedTab(value)}>
                  <Tab
                    label={t('pages.org-details.tabs.users.tabs.individuals', {
                      number: stats.profiles.count,
                    })}
                    value={OrgIndividualsSubtabs.USERS}
                    data-testid="tabUsers"
                  />
                  <Tab
                    label={t('pages.org-details.tabs.users.tabs.pending', {
                      number: stats.pendingInvites.count,
                    })}
                    value={OrgIndividualsSubtabs.INVITES}
                    data-testid="tabInvites"
                  />
                </TabList>
                <Button
                  variant="contained"
                  startIcon={<PersonAddIcon />}
                  onClick={() => navigate('../invite')}
                >
                  {t(
                    'pages.org-details.tabs.users.invite-user-to-organization'
                  )}
                </Button>
              </Box>

              <TabPanel
                value={OrgIndividualsSubtabs.USERS}
                sx={{ px: 0, pt: 0, bgcolor: 'common.white' }}
              >
                <OrgUsersTable
                  orgId={orgId}
                  onChange={async () => {
                    await mutate()
                  }}
                />
              </TabPanel>
              <TabPanel value={OrgIndividualsSubtabs.INVITES} sx={{ px: 0 }}>
                <OrgInvitesTable orgId={orgId} />
              </TabPanel>
            </TabContext>
          </Box>
        </>
      ) : null}
    </Container>
  )
}
