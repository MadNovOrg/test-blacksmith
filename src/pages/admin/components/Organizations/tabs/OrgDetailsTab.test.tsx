import React from 'react'
import { Route, Routes } from 'react-router-dom'

import useOrg, { ProfileType } from '@app/hooks/useOrg'
import { OrgDetailsTab } from '@app/pages/admin/components/Organizations/tabs/OrgDetailsTab'
import { Address, CourseLevel, OfstedRating } from '@app/types'
import { LoadingStatus } from '@app/util'

import { render } from '@test/index'

jest.mock('@app/hooks/useOrg')

const useOrgMock = jest.mocked(useOrg)

describe('OrgDetailsTab', () => {
  useOrgMock.mockReturnValue({
    error: undefined,
    profilesByLevel: new Map<CourseLevel | null, []>(),
    mutate: () => Promise.resolve(undefined),
    status: LoadingStatus.SUCCESS,
    loading: false,
    data: [
      {
        id: '1',
        createdAt: new Date('2020-01-01 00:00:00').toISOString(),
        updatedAt: new Date('2020-01-01 00:00:00').toISOString(),
        name: 'test org',
        tags: [],
        contactDetails: [],
        preferences: {},
        address: {
          line1: 'line 1',
          line2: 'line 2',
          city: 'test city',
          state: 'test state',
          postCode: 'some postcode',
          country: 'test country',
        } as Address,
        attributes: {
          email: 'test@email.com',
          ofstedRating: OfstedRating.INADEQUATE,
          ofstedLastInspection: new Date('2022-01-01 00:00:00').toISOString(),
          localAuthority: 'test authority',
          headFirstName: 'test first name',
          headLastName: 'test last name',
          headTitle: 'test title',
          headPreferredJobTitle: 'test job title',
          website: 'whatever.com',
        },
        sector: 'test sector',
        trustName: 'test trust',
      },
    ],
    stats: {
      '1': {
        profiles: {
          count: 0,
        },
        certificates: {
          active: { count: 0, enrolled: 0 },
          expiringSoon: { count: 0, enrolled: 0 },
          expired: { count: 0, enrolled: 0 },
        },
        pendingInvites: {
          count: 0,
        },
      },
    },
    profiles: [],
    profilesByOrg: new Map<string, ProfileType[]>(),
  })

  it('matches snapshot', async () => {
    const view = render(
      <Routes>
        <Route path="/" element={<OrgDetailsTab orgId="1" />} />
      </Routes>,
      {},
      { initialEntries: ['/'] }
    )

    expect(view).toMatchSnapshot()
  })
})
