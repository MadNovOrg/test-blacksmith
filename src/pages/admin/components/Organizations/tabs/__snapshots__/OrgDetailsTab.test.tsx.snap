// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgDetailsTab matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-w49wrl"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-16q2fz6-MuiTypography-root"
          >
            Organisation details
          </h6>
          <div
            class="MuiBox-root css-12dktrg"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-1xqdqwq-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Name
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    test org
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Organisation email
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    test@email.com
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Sector
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    test sector
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Trust type
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Trust
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    test trust
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Local authority
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    test authority
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Ofsted rating
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    Inadequate
                  </p>
                </div>
                <div
                  class="MuiBox-root css-1coeexk"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                  >
                    Ofsted last inspection
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                  >
                    1 January 2022
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1p35hna-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1bvc4cc"
                >
                  <button
                    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root  css-1vqdlab-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Edit organisation
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-16q2fz6-MuiTypography-root"
          >
            Organisation address
          </h6>
          <div
            class="MuiBox-root css-acjjjk"
          >
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Line 1
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                line 1
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Line 2
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                line 2
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                City
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                test city
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Country
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                test country
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Post code
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                some postcode
              </p>
            </div>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-16q2fz6-MuiTypography-root"
          >
            Additional details
          </h6>
          <div
            class="MuiBox-root css-acjjjk"
          >
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Head first name
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                test first name
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Head last name
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                test last name
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Head title
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                test title
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Head preferred job title
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                test job title
              </p>
            </div>
            <div
              class="MuiBox-root css-1coeexk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
              >
                Website
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
              >
                whatever.com
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-w49wrl"
    >
      <div
        class="MuiBox-root css-0"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-16q2fz6-MuiTypography-root"
        >
          Organisation details
        </h6>
        <div
          class="MuiBox-root css-12dktrg"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-1xqdqwq-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Name
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  test org
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Organisation email
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  test@email.com
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Sector
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  test sector
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Trust type
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Trust
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  test trust
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Local authority
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  test authority
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Ofsted rating
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  Inadequate
                </p>
              </div>
              <div
                class="MuiBox-root css-1coeexk"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
                >
                  Ofsted last inspection
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
                >
                  1 January 2022
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1p35hna-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-1bvc4cc"
              >
                <button
                  class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root  css-1vqdlab-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Edit organisation
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-16q2fz6-MuiTypography-root"
        >
          Organisation address
        </h6>
        <div
          class="MuiBox-root css-acjjjk"
        >
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Line 1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              line 1
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Line 2
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              line 2
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              City
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              test city
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Country
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              test country
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Post code
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              some postcode
            </p>
          </div>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-16q2fz6-MuiTypography-root"
        >
          Additional details
        </h6>
        <div
          class="MuiBox-root css-acjjjk"
        >
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Head first name
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              test first name
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Head last name
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              test last name
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Head title
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              test title
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Head preferred job title
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              test job title
            </p>
          </div>
          <div
            class="MuiBox-root css-1coeexk"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-13eiip-MuiTypography-root"
            >
              Website
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-2a4jea-MuiTypography-root"
            >
              whatever.com
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "context": Object {
    "auth": Object {
      "activeRole": "user",
      "allowedRoles": Set {
        "user",
      },
      "changeRole": [MockFunction],
      "defaultRole": "user",
      "getJWT": [MockFunction],
      "loadProfile": [MockFunction],
      "loading": false,
      "login": [MockFunction],
      "logout": [MockFunction],
      "profile": Object {
        "addresses": Array [],
        "attributes": Array [],
        "avatar": "",
        "contactDetails": Array [],
        "createdAt": "",
        "dietaryRestrictions": null,
        "disabilities": null,
        "dob": "2002-04-06",
        "email": "john.smith@example.com",
        "familyName": "Smith",
        "fullName": "John Smith",
        "givenName": "John",
        "id": "cacb559d-b85d-5e64-b623-37252520ebda",
        "jobTitle": "Developer",
        "lastActivity": 2020-01-01T00:00:00.000Z,
        "organizations": Array [],
        "phone": "+44 1234555",
        "preferences": Array [],
        "roles": Array [
          Object {
            "role": Object {
              "id": "1",
              "name": "user",
            },
          },
        ],
        "tags": Array [],
        "title": "Mr",
        "trainer_role_types": Array [
          Object {
            "trainer_role_type": Object {
              "id": "1",
              "name": "principal",
            },
          },
        ],
        "updatedAt": "",
      },
      "reloadCurrentProfile": [MockFunction],
    },
  },
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
